sonic_data_client/transl_data_client.go

```diff
@@ -454,3 +520,18 @@ func (c *TranslClient) Capabilities() []gnmipb.ModelData {
 func (c *TranslClient) Close() error {
 	return nil
 }
+
+func getBundleVersion(extensions []*gnmi_extpb.Extension) *string {
+	for _,e := range extensions {
+		switch v := e.Ext.(type) {
+			case *gnmi_extpb.Extension_RegisteredExt:
+				if v.RegisteredExt.Id == spb.BUNDLE_VERSION_EXT {
+					var bv spb.BundleVersion
+					proto.Unmarshal(v.RegisteredExt.Msg, &bv)
+					return &bv.Version
```

Comment:
The use_models field is only available in the Get and Subscribe requests, but we needed something for the update/replace/delete requests as well.

Reply:
Ok.. a README on how to use it would be helpful. Looks like spb.BUNDLE_VERSION_EXT is hardcoded and not expressed in proto. So a client cannot figure out how to pass the bundle version just  by looking at the proto file.

Couple of more questions:
1) Can we use ModelData message itself as extension payload?
2) Is it possible to use the standard mechanism for Get and Subscribe, but use the extension only for Set?

Reply:
1) The extension payload can be anything we want, so we may do this. However, I believe the ModelData message is intended for _each_ model, which is why we had a problem with it. The "BundleVersion" was supposed to represent a _global_ versioning of all of the models in a particular sonic release. The indented use of this field is only during the upgrade process and I am not sure anyone else would ever use it.
2) Again I am not sure, since the use_models/ModelData is intended per individual model (I think, I may be wrong) as opposed to the global "BundleVersion". Secondly, I am not sure it would be great to have two different methods to do the same thing in different operations, get,set,subscribe.

The same thing in REST is accomplished via a header "Accept-Version" so we were trying to do something similar to this. The other idea was to send it in the gRPC metadata, but that was not chosen for some reasons I don't recall. The other reason we decided to use the extension is that I was planning on using extensions also for query-parameter like behavior that is to be implemented in REST server for filtering operations etc.